### DISCUSSION TITLE
Alternative to Fuzz

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7633] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

When I use fuzz to lower the file size of my output, I keep getting images like these.

50%
![out 50 fuzz](https://github.com/user-attachments/assets/d5c0998f-7803-42f9-b0bf-4904f5cfc7e9)

25%
![out 25 fuzz](https://github.com/user-attachments/assets/2b38e02f-315c-4ab3-8e63-f4049ea2f489)

Is there an alternative setting when I don't get weird images?

**Comment by snibgo** [comment]
`-fuzz` by itself has no effect on file size. On your thread https://github.com/ImageMagick/ImageMagick/discussions/7623 you said "I don't care about the quality" so I suggested "`-fuzz 25% -layers optimize-transparency`". The fuzz number affects the optimization.

If you don't like that optimization, you might lower the number until it is acceptable, or remove it entirely.

**Reply by Jay123210599** [reply]
Any other ways to optimize it?

**Comment by fmw42** [comment]
What format is the input and what is the desired output?  Your output is PNG. You could use pngcrush or one of the other PNG compression tools at https://imagemagick.org/Usage/formats/#png_non-im

**Reply by Jay123210599** [reply]
the input is animated webp and the output is gif

**Reply by fmw42** [reply]
Compress your webp as much as you like. Then if still too large, optimize the gif and also reduce colors. Those are the only ways that I know to reduce gif size other than resizing smaller dimensions.

**Reply by Jay123210599** [reply]
how do I use imagemagick to reduce the colors?

**Reply by fmw42** [reply]
+dither -colors N where N is between 0 and 256.  Or in IM 7, you can also use -kmeans N.  See the Select CLI Option selector at the top of https://imagemagick.org/script/command-line-options.php
